insonmnia/hub/miner_ctx.go

```diff
@@ -58,16 +60,24 @@ type MinerCtx struct {
 }
 
 func (h *Hub) createMinerCtx(ctx context.Context, conn net.Conn) (*MinerCtx, error) {
+	var err error
+
+	if h.creds != nil {
+		conn, err = h.tlsHandshake(ctx, conn)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	var (
 		m = MinerCtx{
 			conn:         conn,
 			statusMap:    make(map[string]*pb.TaskStatusReply),
 			usageMapping: make(map[OrderId]*resource.Resources),
 		}
-		err error
 	)
 	m.ctx, m.cancel = context.WithCancel(ctx)
-	m.grpcConn, err = util.MakeGrpcClient(ctx, "miner", h.creds, grpc.WithDialer(func(_ string, _ time.Duration) (net.Conn, error) {
+	m.grpcConn, err = util.MakeGrpcClient(ctx, "miner", nil, grpc.WithDialer(func(_ string, _ time.Duration) (net.Conn, error) {
```

Comment:
Why `util.MakeGrpcClient` receive "miner" as address parameter?

Reply:
Nothing will change if we replace "miner" with ""

Reply:
![](https://memegenerator.net/img-preview/Instance/PreviewInstanceData?instanceDataJson=%7B%22instanceElements%22%3A%5B%7B%22instancePicture%22%3A%7B%22imageID%22%3A166088%7D%2C%22instanceTexts%22%3A%5B%7B%22text%22%3A%22%20%20%20%20%20%20%20%20%20%22%2C%22placeholder%22%3A%22top%20text%22%2C%22horizontalAlign%22%3A%22center%22%2C%22verticalAlign%22%3A%22near%22%2C%22color%22%3A%22%23ffffff%22%2C%22fontSize%22%3A%221%22%7D%2C%7B%22text%22%3A%22Y%20U%20receive%20%5C%22miner%5C%22%20as%20address%22%2C%22placeholder%22%3A%22bottom%20text%22%2C%22horizontalAlign%22%3A%22center%22%2C%22verticalAlign%22%3A%22far%22%2C%22color%22%3A%22%23ffffff%22%7D%5D%7D%5D%7D)